server/jetstream_api.go

```diff
@@ -3066,6 +3041,81 @@ func isEmptyRequest(req []byte) bool {
 	return len(vm) == 0
 }
 
+// getStepDownPreferredPlacement attempts to work out what the best placement is
+// for a stepdown request. The preferred server name always takes precedence, but
+// if not specified, the placement will be used to filter by cluster. The caller
+// should check for return API errors and return those to the requestor if needed.
+func (s *Server) getStepDownPreferredPlacement(group RaftNode, preferredServer string, placement *Placement) (string, *ApiError) {
+	var preferredLeader string
+	if preferredServer != _EMPTY_ {
+		for _, p := range group.Peers() {
+			si, ok := s.nodeToInfo.Load(p.ID)
+			if !ok || si == nil {
+				continue
+			}
+			if si.(nodeInfo).name == preferredServer {
+				preferredLeader = p.ID
+				break
+			}
+		}
+		if preferredLeader == group.ID() {
+			return _EMPTY_, NewJSClusterNoPeersError(fmt.Errorf("preferred server %q is already leader", preferredServer))
+		}
+		if preferredLeader == _EMPTY_ {
+			return _EMPTY_, NewJSClusterNoPeersError(fmt.Errorf("preferred server %q not known", preferredServer))
```

Comment:
Could this be a way for folks to poke and discover all the server names?

Reply:
I don't think this is any worse than, say, creating and deleting lots of replicated streams and looking at the stream infos to see where the replicas are.

Reply:
ok